src/components/BoardHeader/HeaderMenu/HeaderMenu.test.tsx

```diff
@@ -1,43 +1,129 @@
+
+jest.mock("store", () => ({
+  ...jest.requireActual("store"),
+  dispatch: jest.fn(),
+}));
+
+jest.mock("utils/export", () => ({
+  ...jest.requireActual("utils/export"),
+  exportAsJSON: jest.fn(),
+  exportAsCSV: jest.fn(),
+  exportAsCSVZip: jest.fn(),
+}));
+
+jest.mock("file-saver", () => ({saveAs: jest.fn()}));
+
+const initialState = {
+  board: {
+    data: {
+      name: "board",
+      showAuthors: true,
+      joinConfirmationRequired: true,
+      id: "boardId",
     },
+  },
+};
+
+const createHeaderMenu = (currentUserIsModerator: boolean) => {
+  const store = mockStore(initialState);
+  const [HeaderMenuContext] = wrapWithTestBackend(HeaderMenu);
+  return (
+    <Provider store={store}>
+      <HeaderMenuContext open onClose={() => undefined} currentUserIsModerator={currentUserIsModerator} />
+    </Provider>
+  );
+};
+
+describe("HeaderMenu", () => {
+  beforeEach(() => {
+    const portal = global.document.createElement("div");
+    portal.setAttribute("id", "portal");
+    global.document.querySelector("body")!.appendChild(portal);
   });
 
   describe("should render correctly", () => {
-    test("on open", () => {
-      const portal = global.document.createElement("div");
-      portal.setAttribute("id", "portal");
-      global.document.querySelector("body")!.appendChild(portal);
-
-      const {container} = render(
-        <Provider store={store}>
-          <HeaderMenu open onClose={() => undefined} />
-        </Provider>,
-        {container: global.document.querySelector("#portal")!}
-      );
+    test("moderator", () => {
+      const {container} = render(createHeaderMenu(true), {container: global.document.querySelector("#portal")!});
       expect(container).toMatchSnapshot();
     });
 
-    test("on close", () => {
-      const {container} = render(
-        <Provider store={store}>
-          <HeaderMenu open={false} onClose={() => undefined} />
-        </Provider>,
-        {container: global.document.querySelector("#portal")!}
-      );
+    test("user", () => {
+      const {container} = render(createHeaderMenu(false), {container: global.document.querySelector("#portal")!});
       expect(container).toMatchSnapshot();
     });
+
+    test("Click on hide authors", () => {
+      const {container} = render(createHeaderMenu(true), {container: global.document.querySelector("#portal")!});
+      const button = container.querySelector(".header-menu")?.children.item(1)?.firstChild?.firstChild?.firstChild!;
+      expect(button).toHaveClass("menu__item-button");
+      fireEvent.click(button);
+      expect(store.dispatch).toHaveBeenCalledWith(ActionFactory.editBoard({id: "boardId", showAuthors: false}));
+    });
+
+    test("Click on hide notes", () => {
+      const {container} = render(createHeaderMenu(true), {container: global.document.querySelector("#portal")!});
+      const button = container.querySelector(".header-menu")?.children.item(1)?.children.item(1)?.firstChild?.firstChild!;
```

Comment:
Please try to query for more specific elements. `children.item(1)?.children.item(1)?.firstChild?.firstChild!` will break very fast once the dom structure changes :)